src/modules/vision_target_estimator/Orientation/KF_orientation_moving.cpp

```diff
@@ -0,0 +1,132 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2013-2018 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file KF_orientation_moving.cpp
+ * @brief Filter to estimate the orientation of moving targets. State: [theta, theta_dot]
+ *
+ * @author Jonas Perolini <jonspero@me.com>
+ *
+ */
+
+#include "KF_orientation_moving.h"
+
+namespace vision_target_estimator
+{
+
+void KF_orientation_moving::predictState(float dt)
+{
+	/*
+	⎡dt⋅theta_dot + theta⎤
+	⎣      theta_dot     ⎦
+	*/
+
+	_state(0, 0) = _state(0, 0) + _state(1, 0) * dt;
```

Comment:
I think it might be more readable if you either make a struct for the state as we do in ekf2 (then you can use it as`_state.yaw + _state.yaw_rate * dt`) or save the index in an enum (`_state(State::yaw) = ....`).

Reply:
Mostly done in https://github.com/PX4/PX4-Autopilot/pull/22008/commits/d03d669227263d86c649d7f13bb55a5c3677a3f0. Still work to be done for the measurement matrix.